src/main/java/seedu/address/logic/parser/FindByNricCommandParser.java

```diff
@@ -0,0 +1,28 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.logic.commands.FindByNricCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.NricContainsKeywordPredicate;
+
+/**
+ * Parses input arguments and creates a new FindByNricCommand object
+ */
+public class FindByNricCommandParser implements Parser<FindByNricCommand> {
+
+    /**
+     * Parses the given {@code String} argument in the context of the FindByNricCommand
+     * and returns a FindByNricCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindByNricCommand parse(String args) throws ParseException {
+        String trimmedArg = args.trim();
+        if (trimmedArg.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindByNricCommand.MESSAGE_USAGE));
+        }
```

Comment:
Add checks for invalid nric inputs as well.
Right now `find_nric t1234567` `find_nric 1234567` `find_nric 1234567s` still works and finds 0 patients

Also add checks for starting letter to be S or T and 1234567 are numbers

Update tests accordingly